src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -40,4 +40,34 @@ export class LocalTaskService implements TaskService {
   private writeTasks(tasks: Task[]): void {
     localStorage.setItem(LocalTaskService.STORAGE_KEY, JSON.stringify(tasks));
   }
+  editTask(task: Task): any {
+    const tasks = this.readTasks();
+    const item = tasks.find(x => x.id === task.id);
+    if (item){
+      item.status = 'done';
+    }
+    localStorage.setItem(LocalTaskService.STORAGE_KEY, JSON.stringify(tasks));
+    this.sortTasks();
```

Comment:
why not sort them before storing instead of storing the items, then reading them again, sorting and storing yet again?